### PR TITLE
LPD-27022 Enable CX workflow action even if the task doesn't have a user as assignee

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-27022

### CODE REVIEW

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]
As we discussed, can you bring back the protected `doExecute` method?

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImpl.java [hunk 1]
```diff
@@ -171,42 +172,47 @@ else if (jsonObject.has("serializable")) {
 					"There are no workflow task assignees for Kaleo action " +
```
Comment (guilhermedcamacho) [inline modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImpl.java:172]: Do we still need this condition just to log this? Maybe removing this `return;` statement will make this unusable.

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImpl.java [hunk 2]
```diff
@@ -171,42 +172,47 @@ else if (jsonObject.has("serializable")) {
...
+
+	protected void launch(
+			JSONObject payloadJSONObject,
+			List<WorkflowTaskAssignee> workflowTaskAssignees)
+		throws Exception {
+
 		WorkflowTaskAssignee workflowTaskAssignee = workflowTaskAssignees.get(
 			0);
 
+		long userId = workflowTaskAssignee.getAssigneeClassPK();
```
Comment (guilhermedcamacho) [inline modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImpl.java:197]: I think that if the `workflowTaskAssignees` comes empty, the `workflowTaskAssignee` variable will be null and then we will have an NPE at this point, am I right?

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImpl.java [hunk 3]
```diff
@@ -171,42 +172,47 @@ else if (jsonObject.has("serializable")) {
...
 		if (!Objects.equals(
 				workflowTaskAssignee.getAssigneeClassName(),
 				User.class.getName())) {
 
-			if (_log.isWarnEnabled()) {
-				_log.warn(
-					"The first workflow task assignee must be a user for " +
-						"Kaleo action " + kaleoAction);
-			}
+			User defaultServiceAccountUser =
```
Comment (guilhermedcamacho) [inline modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImpl.java:203]: I think you can call  `_userLocalService.getUserIdByScreenName` method and inline this below.

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/test/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImplTest.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		UserLocalService userLocalService = Mockito.mock(
+			UserLocalService.class);
+
+		long companyId = RandomTestUtil.randomLong();
+
+		ReflectionTestUtil.setFieldValue(
+			functionActionExecutorImpl, "_companyId", companyId);
+
+		Mockito.when(
+			userLocalService.getUserByScreenName(
```
Comment (guilhermedcamacho) [inline modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/test/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImplTest.java:61]: If you do what I suggested before, please remember to change this to `userLocalService.getUserIdByScreenName`

File: modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/test/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImplTest.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+			functionActionExecutorImpl,
+			"_functionActionExecutorImplConfiguration",
+			functionActionExecutorImplConfiguration);
+
+		PortalCatapult portalCatapult = Mockito.mock(PortalCatapult.class);
+
+		ReflectionTestUtil.setFieldValue(
+			functionActionExecutorImpl, "_portalCatapult", portalCatapult);
+
+		List<WorkflowTaskAssignee> workflowTaskAssignees =
```
Comment (guilhermedcamacho) [inline modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/test/java/com/liferay/portal/workflow/kaleo/runtime/internal/action/executor/FunctionActionExecutorImplTest.java:96]: If the workflowTaskAssignees list could be null, remember to cover this scenario if possible.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by guilhermedcamacho — APPROVED** [review]

**Comment by guilhermedcamacho** [comment]
ci:test:sf

**Comment by guilhermedcamacho** [comment]
ci:test:relevant

**Comment by guilhermedcamacho** [comment]
ci:stop

**Comment by guilhermedcamacho** [comment]
SF error fixed in #1173

**Pull request closed without merging**